Tests/EmailValidator/Validation/NonLocalValidationTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Egulias\Tests\EmailValidator\Validation;
+
+use Egulias\EmailValidator\EmailLexer;
+use Egulias\EmailValidator\Validation\Error\LocalEmail;
+use Egulias\EmailValidator\Validation\NonLocalValidation;
+use PHPUnit\Framework\TestCase;
+
+class NonLocalValidationTest extends TestCase
+{
+    public function validEmailsProvider()
+    {
+        return [
+            ['test@localhost.org'],
+            ['test.test@example.org'],
+        ];
+    }
+
+    public function invalidEmailsProvider()
+    {
+        return [
+            ['test@localhost'],
+            ['test.test@example'],
+        ];
+    }
+
+    /**
+     * @dataProvider validEmailsProvider
+     */
+    public function testValidNonLocal($validEmail)
+    {
+        $validation = new NonLocalValidation();
+        $this->assertTrue($validation->isValid($validEmail, new EmailLexer()));
+    }
+
+    /**
+     * @dataProvider invalidEmailsProvider
+     */
+    public function testInvalidNonLocal($invalidEmail)
+    {
+        $validation = new NonLocalValidation();
+        $this->assertFalse($validation->isValid($invalidEmail, new EmailLexer()));
+        
+        $expectedError = new LocalEmail();
+        $this->assertEquals($expectedError, $validation->getError());
```

Comment:
This assertion I think is not correct to assert the expected error message is same.
I think it should assert the two error messages are same.

Reply:
assertEquals checks inf content (property values) equals instead of same object reference i think?
issame checks for same object reference, so in that case this should be fine? 

(i would do an instance of check, since thats what you really want i think?)

Reply:
I think it should use the `assertInstanceOf` to check whether the class instance is `LocalEmail` class here, not using `assertEquals`.

Reply:
After tracing `NonLocalValidation` class, the `$this->error` will be the bool value or `LocalEmail` class instance.

It should use the different assertion for asserting the different `$this->error` value.
Consider following code:

```php

$this->assertInstanceOf(LocalEmail::class, $validation->getError());

$this->assertTrue($validation->getError());

$this->assertFalse($validation->getError());
```

Reply:
Very true, fixed the instanceof check.